src/utils/location.ts

```diff
@@ -10,6 +10,16 @@ export function addQueryParam(location: Location, key: string, value: string): s
   return `${location.pathname}${searchParams ? `?${searchParams}` : ''}`;
 }
 
+export function replaceQueryParam(location: Location, key: string, value: string): string {
+  const urlSearchParams = new URLSearchParams();
+
+  urlSearchParams.set(key, value);
+
+  const searchParams = urlSearchParams.toString();
+
+  return `${location.pathname}?${searchParams}`;
+}
+
```

Comment:
I initially thought this was the same function as `addQueryParam`, but this function clears all existing query params and only adds the given param. Let's add a comment explaining this.